src/core-layers/screen-grid-layer/screen-grid-layer.js

```diff
@@ -56,37 +65,104 @@ export default class ScreenGridLayer extends Layer {
     }
     /* eslint-disable max-len */
     attributeManager.addInstanced({
-      instancePositions: {size: 3, update: this.calculateInstancePositions},
-      instanceColors: {
-        size: 4,
-        type: GL.UNSIGNED_BYTE,
-        transition: true,
-        accessor: ['getPosition', 'getWeight'],
-        update: this.calculateInstanceColors
-      }
+      instancePositions: {size: 3, update: this.calculateInstancePositions}
     });
     /* eslint-disable max-len */
 
-    this.setState({model: this._getModel(gl)});
+    const options = {
+      id: `${this.id}-aggregator`,
+      shaderCache: this.context.shaderCache
+    };
+    this.setState({
+      model: this._getModel(gl),
+      screenGridAggregator: new ScreenGridAggregator(gl, options)
+    });
   }
 
   shouldUpdateState({changeFlags}) {
     return changeFlags.somethingChanged;
   }
 
+  updateAttribute({props, oldProps, changeFlags}) {
+    const {gl} = this.context;
+    if (LOG_LEVEL > 0) {
+      console.log(
+        `Switching from ${oldProps.gpuAggregation ? 'GPU' : 'CPU'} to ${
+          props.gpuAggregation ? 'GPU' : 'CPU'
+        }`
+      );
+    }
+    const attributeManager = this.getAttributeManager();
+
+    if (props.gpuAggregation) {
+      if (LOG_LEVEL > 0) {
+        console.log(`remove instanceColors atatribute`);
+      }
+      attributeManager.remove(['instanceColors']);
+    } else {
+      if (LOG_LEVEL > 0) {
+        console.log(`add instanceColors atatribute`);
+      }
+      attributeManager.addInstanced({
+        instanceColors: {
+          size: 4,
+          type: GL.UNSIGNED_BYTE,
+          transition: true,
+          accessor: ['getPosition', 'getWeight'],
+          update: this.calculateInstanceColors
+        }
+      });
+    }
+    this.setState({model: this._getModel(gl)});
+    attributeManager.invalidateAll();
+  }
+
   updateState({oldProps, props, changeFlags}) {
+    const {gl} = this.context;
     super.updateState({props, oldProps, changeFlags});
     const cellSizeChanged = props.cellSizePixels !== oldProps.cellSizePixels;
+    const aggregationChanged =
+      ScreenGridAggregator.hasGPUSupport(gl) && props.gpuAggregation !== oldProps.gpuAggregation;
```

Comment:
is it possible ScreenGridAggregator.hasGPUSupport(gl) itself got changed (e.g. graphics card being used got changed due to power mode change)? if yes, shall we consider aggregationChanged === true as well?